src/target/riscv/riscv-013.c

```diff
@@ -2298,6 +2340,127 @@ static int batch_run(const struct target *target, struct riscv_batch *batch)
 	return riscv_batch_run(batch);
 }
 
+/*
+ * Performs a memory read using memory access abstract commands. The read sizes
+ * supported are 1, 2, and 4 bytes despite the spec's support of 8 and 16 byte
+ * aamsize fields in the memory access abstract command.
+ */
+static int read_memory_abstract(struct target *target, target_addr_t address,
+		uint32_t size, uint32_t count, uint8_t *buffer)
+{
+	int result = ERROR_OK;
+
+	LOG_DEBUG("reading %d words of %d bytes from 0x%" TARGET_PRIxADDR, count,
+			  size, address);
+
+	memset(buffer, 0, count*size);
+
+	// Convert the size (bytes) to width (bits)
+	unsigned width = size << 3;
+	if (width > 64) {
+		// TODO: Add 128b support if it's ever used. Involves modifying
+		//       read/write_abstract_arg() to work on two 64b values.
+		LOG_ERROR("Unsupported size: %d bits", size);
+		return ERROR_FAIL;
+	}
+
+	// Create the command (physical address, postincrement, read)
+	uint32_t command = access_memory_command(target, false, width, true, false);
```

Comment:
Both `read_memory_abstract()` and `write_memory_abstract()` build their access memory command using `aampostincrement = true`. My reading of the 0.14 debug spec indicates that access memory abstract commands are optional, but that none of the fields are optional. This is in contrast to the access register abstract command where `aarpostincrement` is specifically stated as being optional.

Can anyone confirm my reading, or should I create some probe code which tries to determine if the support for `aampostincrement` for the access memory abstract command?

Reply:
aampostincrement is optional, as are all other arguments. The intro to "Abstract Commands" says: "Commands may be supported with some options set, but not with other options set."

Probe code would be great, but I'm happy to accept that we only support abstract memory access when aampostincrement is supported for now. We can add support for targets that don't support it when we encounter them.

Reply:
Thanks. I'll leave it to that future target then, but I'm certainly happy to help write the probe code when that target arrives. I just don't want to write and push untested code.